### PR TITLE
Run app in docker

### PR DESCRIPTION
**Pull request by tonysimmers002**
Christmas leave and I spend it doing almost work stuff?!  
Actually this is the first pull request / collaboration in github for me so be gentle.  If you don't know me I'm a mate of Cory's working at MSD -- he thought this might be a step towards hosting on AWS (or similar)

Changes are to wrap the app into a Docker container.  No changes to the app itself, just added a docker folder with a few files.   Assuming you have docker and docker-compose you should be able to create and run a container by going into that folder and
   ```
docker-compose build     # gets rocker/shiny-verse:4.0, uses devtools to install app, and sets up a shiny server
   APP_PORT=6789 docker-compose up -d    # starts it on localhost:6789

### CODE REVIEW

**Comment by kcha193** [comment]
Looks good to me, so does it means that I don't need to host it on my shinyapp.io account anymore? i.e. https://kcha193.shinyapps.io/IncomeExplorer/

**Comment by gentleboil** [comment]
Thanks Kevin, it will be a while before there is a public host of that
container....
I liked your moving the scenarios in with the app.  I wasn't sure if, or
more to the point how, I should have merged in your branch to mine.

**Comment by kcha193** [comment]
Hi Tony, be my guest to do whatever, although I would wait till my branch gets merged to master first via the other PR. 

I have other plans in making the UI and plots look better, but I will do it in my own repo first. 

Have a good one, 

Kevin

**Comment by corypdavis** [comment]
Hi guys, sorry for the slow response on this, and thanks so much @gentleboil  and @kcha193 for your time on this.  Have added Luke Yvonne and I as reviewers; we'll aim to have a chat shortly.  This a great capability to add, and it is completely orthogonal to everything else, so I see no problem in merging it.  It has been a while since I used docker though so potentially a comment at the top of each file indicating its purpose would be handy.

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+version: '3.5'
```
Comment (corypdavis) [inline docker/docker-compose.yml:1]: maybe just a comment here indicating what this file actually does?

Comment (corypdavis) [reply docker/docker-compose.yml:1]: maybe more generally indicating how these four files work together - in the README file?

**Review by corypdavis — APPROVED** [review]
Thanks so much for this @gentleboil  - the comments are just suggestions that could make things clearer for non-Docker experts.

**Review by Luke-Symes-Tsy — APPROVED** [review]

File: docker/docker-compose.yml (hunk 1 above)
Comment (tonysimmers002) [reply docker/docker-compose.yml:1]: Yes - I had realised it needed something more in the documentation.  Of course I have now forgotten what I did and why...I'll do that over the next few days.

**Pull request merged**